### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by paigekato**
### Small Improvements 
- Added ability to have optional ip address instead
- Updated readme to reflect changes and remove irrelevant stuff
- Add error handling from API - prints errors to `craft/storage/runtime/logs/geo.log`
- Add stored ip data cookie name to config

### CODE REVIEW

File: geo/services/Geo_LocationService.php [hunk 1]
```diff
@@ -25,12 +23,12 @@ public function getIsEu()
...
 	
-	private function fetchIpData()
+	private function fetchIpData($ip)
 	{
 		$devMode = craft()->config->get('devMode');
 		$ipApiKey = craft()->config->get('ipApiKey', 'geo');
-		$ip = craft()->request->getIpAddress();
-		$isEuUserCookie = 'craft_geo_eu_user';
+		$ip = $ip ? $ip : craft()->request->getIpAddress();
+		$isEuUserCookie = craft()->config->get('locationDataCookie', 'geo');
```
Comment (dstello) [inline geo/services/Geo_LocationService.php]: `isEuUserCookie` :) still referencing EU

File: geo/variables/GeoVariable.php [hunk 2]
```diff
@@ -3,13 +3,13 @@
 
 class GeoVariable
   {
-  public function data()
+  public function data($ip='')
```
Comment (dstello) [inline geo/variables/GeoVariable.php:6]: i think we want to move the default to `fetchIpData`

Comment (dstello) [reply geo/variables/GeoVariable.php:6]: right? unless you are getting an error the other way?

Comment (paigekato) [reply geo/variables/GeoVariable.php:6]: I think it needs to stay here. It is throwing an error the other way

Comment (dstello) [reply geo/variables/GeoVariable.php:6]: neat

**Comment by dstello** [comment]
lgts

**Pull request merged**